### PR TITLE
:bug: Prevent circular parent-child task relationships

### PR DESCRIPTION
**Pull request by fabianvf**
Also allows children to maintain their original priority level (for sorting) while remaining in the queue of their oldest ancestor

### CODE REVIEW

File: kai/reactive_codeplanner/main.py [hunk 1]
```diff
@@ -16,17 +19,26 @@
...
     MavenCompileStep,
 )
+from kai.reactive_codeplanner.task_runner.dependency.task_runner import (
+    DependencyTaskRunner,
+)
 from kai.reactive_codeplanner.vfs.git_vfs import RepoContextManager
 
 # Configure logging
-logging.basicConfig(level=logging.INFO)
+logging.basicConfig(level=logging.DEBUG)
```
Comment (shawn-hurley) [inline kai/reactive_codeplanner/main.py:28]: Why this change?

File: kai/reactive_codeplanner/task_runner/analyzer_lsp/api.py [hunk 2]
```diff
@@ -18,6 +18,11 @@ class AnalyzerRuleViolation(ValidationError):
     # TODO Highest priority?
     priority: int = 2
 
+    def __str__(self) -> str:
+        return f"{self.__class__.__name__}<loc={self.file}:{self.line}:{self.column}, message={self.violation.description}>"
```
Comment (shawn-hurley) [inline kai/reactive_codeplanner/task_runner/analyzer_lsp/api.py:22]: What do you think of adding the ruleID that triggered it as well?

File: kai/reactive_codeplanner/task_runner/analyzer_lsp/validator.py [hunk 3]
```diff
@@ -156,7 +156,7 @@ def __parse_analyzer_lsp_output(
                         class_to_use = AnalyzerDependencyRuleViolation
                     validation_errors.append(
                         class_to_use(
-                            file=urlparse(i.uri).path,
+                            file=urlparse(i.uri).path.lstrip("/"),
```
Comment (shawn-hurley) [inline kai/reactive_codeplanner/task_runner/analyzer_lsp/validator.py:159]: Why this change?

**Review by shawn-hurley — APPROVED** [review]
Assuming we are passing CI, I am good I just have questions for my own understanding

File: kai/reactive_codeplanner/main.py (hunk 1 above)
Comment (fabianvf) [reply kai/reactive_codeplanner/main.py:28]: Cruft from before I rebased on the logging PR

File: kai/reactive_codeplanner/task_runner/analyzer_lsp/validator.py (hunk 3 above)
Comment (fabianvf) [reply kai/reactive_codeplanner/task_runner/analyzer_lsp/validator.py:159]: I was getting relative paths starting with a slash (ie, `/example/coolstore/src/...`), I just had this in to fix it though I did mean to follow up on it. Can remove from this PR, also curious if anyone else is seeing that

File: kai/reactive_codeplanner/main.py (hunk 1 above)
Comment (shawn-hurley) [reply kai/reactive_codeplanner/main.py:28]: I think we may just want to remove this line then?

File: kai/reactive_codeplanner/task_runner/analyzer_lsp/validator.py (hunk 3 above)
Comment (shawn-hurley) [reply kai/reactive_codeplanner/task_runner/analyzer_lsp/validator.py:159]: we should write an issue in analyzer and track that down

Comment (fabianvf) [reply kai/reactive_codeplanner/task_runner/analyzer_lsp/validator.py:159]: Should I drop it here? Nothing works for me without this change

Comment (shawn-hurley) [reply kai/reactive_codeplanner/task_runner/analyzer_lsp/validator.py:159]: No we should keep it, but I just want to create a follow on issue because IIRC this was a conversation with the IDE/UX folks and I want to make sure that we track it. 

I am good with keeping just wanted to know :)

**Review by shawn-hurley — APPROVED** [review]
/lgtm

File: kai/reactive_codeplanner/task_manager/api.py [hunk 4]
```diff
@@ -29,6 +29,11 @@ class Task:
 
     _creation_counter = 0
 
+    def oldest_ancestor(self) -> "Task":
+        if self.parent:
+            return self.parent.oldest_ancestor()
+        return self
+
```
Comment (JonahSussman) [inline kai/reactive_codeplanner/task_manager/api.py:36]: Small performance thing. Are `Task`s immutable? If so, we could make this a cached property.

**Review by JonahSussman — APPROVED** [review]

File: kai/reactive_codeplanner/task_manager/api.py (hunk 4 above)
Comment (fabianvf) [reply kai/reactive_codeplanner/task_manager/api.py:36]: hmm, they are not immutable, but the parent-child relationships are AFAIK. Though it'd probably save us a microsecond :joy:

Comment (JonahSussman) [reply kai/reactive_codeplanner/task_manager/api.py:36]: Hey those microseconds add up 👀
